src/matches.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+export type NormalizerFn = (textToNormalize: string) => string;
+
+function matches(
+  textToMatch: string,
+  matcher: string | RegExp,
+  normalizer: NormalizerFn,
+  exact?: boolean = true
+) {
```

Comment:
```suggestion
function matches(
  matcher: string | RegExp,
  text: string,
  normalizer: NormalizerFn,
  exact?: boolean = true
) {
```

readability, nit: more natural order would be "matcher matches text", hence the suggested argument order. 

Also textToMatch felt confusing as I thought that this is the text we are looking for. Suggested rename: `text` or `textToMatchAgainst`, with `text` being simplier.